spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulServerList.java

```diff
@@ -83,20 +85,24 @@ public void initWithNiwsConfig(IClientConfig clientConfig) {
 		if (response.getValue() == null || response.getValue().isEmpty()) {
 			return Collections.emptyList();
 		}
-		return transformResponse(response.getValue());
+		return transformResponse(response.getValue(),
+				this.properties.getStatusConsideredAsHealthy());
 	}
 
 	/**
 	 * Transforms the response from Consul in to a list of usable {@link ConsulServer}s.
 	 * @param healthServices the initial list of servers from Consul. Guaranteed to be
 	 * non-empty list
+	 * @param statusConsideredAsHealthy list of status that are considered as
+	 * healthy/passing
 	 * @return ConsulServer instances
-	 * @see ConsulServer#ConsulServer(HealthService)
+	 * @see ConsulServer#ConsulServer(HealthService, java.util.Set)
 	 */
-	protected List<ConsulServer> transformResponse(List<HealthService> healthServices) {
+	protected List<ConsulServer> transformResponse(List<HealthService> healthServices,
```

Comment:
To avoid changing the method signature I would use `this.properties` in the method rather than pass in the `Set`.  You can add an overloaded method if you want to give the option to past in the `Set`